docs/workflow/ci/triaging-failures.md

```diff
@@ -0,0 +1,137 @@
+dotnet tool update --global runfo
+```
+
+Then use the `runfo get-helix-payload` command.
+
+Note that if a test fails and produces a core (crash) dump, the Azure DevOps "Artifacts" page will include a link to
+the crash dump as well as a `how-to-debug-dump.md` file that describes using `runfo` to download the assets and other
+tools to do the debugging. This file is built from the template [here](../../../eng/testing/debug-dump-template.md).
+
+## Determining if a failure is due to JIT stress
+
+To extensively test the CLR JIT, there are many automated test runs using the coreclr and libraries tests
+but setting additional configuration settings. Because tests are run in so many JIT stress configurations,
+any flakiness in a test is much more likely to occur in a JIT stress run and not necessarily be due to the
+JIT stress mode itself. This section describes how to determine if a failure is likely due to a JIT bug.
+
+### Source of test configurations
+
+The mapping from Azure DevOps pipeline to configuration settings is set by the `scenarios` tags for coreclr
+tests [here](../../../eng/pipelines/common/templates/runtimes/run-test-job.yml) and for libraries tests
+[here](../../../eng/pipelines/libraries/run-test-job.yml).
+These tags are converted to configuration variables [here](../../../src/tests/Common/testenvironment.proj).
+
+However, all you really need to know is the set of environment variables set, as described above.
+
+### Asserts
+
+Of course, if the failure is due to a JIT assertion failure, the problem is obviously a bug in the JIT.
+
+An example:
+
+```
+Assert failure(PID 61 [0x0000003d], Thread: 75 [0x004b]): Assertion failed 'node->DefinesLocalAddr(this, size, &lcl, nullptr) && lvaGetDesc(lcl)->lvHiddenBufferStructArg' in 'System.Reflection.TypeLoading.Ecma.EcmaEvent:ComputeEventRaiseMethod():System.Reflection.TypeLoading.RoMethod:this' during 'Morph - Global' (IL size 37; hash 0x941ee672; FullOpts)
+
+    File: /__w/1/s/src/coreclr/jit/gentree.cpp Line: 17913
+    Image: /root/helix/work/correlation/dotnet
+```
+
+Note, in particular, the `File:` line includes the path `src/coreclr/jit`.
+
+### Repro without JIT stress modes
+
+For other failures, first attempt to reproduce as it failed in the CI, with the same configuration settings. For intermittent failures,
+this might require running the test in a loop, applying artificial load to the machine, running on the exact architecture/OS/OS version/Docker container, etc.
+
+Once the problem can be reproduced, attempt to reproduce the problem without setting any of the JIT stress variables, e.g., do not set:
+- `COMPlus_TieredCompilation`
+- `COMPlus_JitStress`
+- `COMPlus_JitStressRegs`
+
+If the test reliably fails with the JIT stress modes, but passes without, consider it a JIT issue.
+
+### Intermittent test failures
+
+If it is difficult to reproduce an issue, a useful tool is to use [Kusto](https://dataexplorer.azure.com/clusters/engsrvprod/databases/engineeringdata)
```

Comment:
When a test fails in Jitstress, would it make sense to ask the opener to run and paste in the results of a canned Kusto query to see whether it's only failing in Jitstress? Then they could decide whether to send it to codegen team first or not. (Of course, it may well be a timing issue, but presumably this heuristic would be more accurate than always sending to the area owners first)

Reply:
It might be interesting to do a hand sampling of the last 20-50 bugs marked jit stress to see what proportion were actually test issues (not codegen team issues), and of those how many were only failing in jitstress. That could help us tune our process so they go to the right team first.

Reply:
Yes, this is what I tend to use to first rule it out. No need to tag jit stress if it reproduces obviously outside the range (I say a significant portion because one or two failures outside usually hints at other issues like a run with something more systemic and less likely to be the test itself).